pkg/filetree/glob.go

```diff
@@ -0,0 +1,192 @@
+	fn, err := a.filetree.node(file.Path(name), linkResolutionStrategy{
+		FollowAncestorLinks: true,
+		// Lstat by definition requires that basename symlinks are not followed
+		FollowBasenameLinks:          false,
+		DoNotFollowDeadBasenameLinks: false,
+	})
+	if err != nil {
+		return &fileinfoAdapter{}, err
+	}
+	if fn == nil {
+		return &fileinfoAdapter{}, os.ErrNotExist
+	}
+
+	return &fileinfoAdapter{
+		VirtualPath: file.Path(name),
+		Node:        *fn,
+	}, nil
+}
+
+// Open the given file path and return a doublestar.File.
+func (a *osAdapter) Open(name string) (doublestar.File, error) {
+	return &fileAdapter{
+		os:       a,
+		filetree: a.filetree,
+		name:     name,
+	}, nil
+}
+
+// PathSeparator returns the standard separator between path entries for the underlying filesystem.
+func (a *osAdapter) PathSeparator() rune {
+	return []rune(file.DirSeparator)[0]
+}
+
+// Stat returns a FileInfo describing the named file.
+func (a *osAdapter) Stat(name string) (os.FileInfo, error) {
+	fn, err := a.filetree.node(file.Path(name), linkResolutionStrategy{
+		FollowAncestorLinks:          true,
+		FollowBasenameLinks:          true,
+		DoNotFollowDeadBasenameLinks: a.doNotFollowDeadBasenameLinks,
+	})
+	if err != nil {
+		return &fileinfoAdapter{}, err
+	}
+	if fn == nil {
+		return &fileinfoAdapter{}, os.ErrNotExist
+	}
+	return &fileinfoAdapter{
+		VirtualPath: file.Path(name),
+		Node:        *fn,
+	}, nil
+}
+
+// fileinfoAdapter is meant to implement the os.FileInfo interface intended only for glob searching. This does NOT
+// report correct metadata for all behavior.
+type fileinfoAdapter struct {
+	VirtualPath file.Path
+	Node        filenode.FileNode
+}
+
+// Name base name of the file
+func (a *fileinfoAdapter) Name() string {
+	return a.VirtualPath.Basename()
+}
+
+// Size is a dummy return value (since it is not important for globbing). Traditionally this would be the length in
+// bytes for regular files.
+func (a *fileinfoAdapter) Size() int64 {
+	return 1
+}
+
+// Mode returns the file mode bits for the given file. Note that the only important bits in the bitset is the
+// dir and symlink indicators; no other values can be used.
+func (a *fileinfoAdapter) Mode() os.FileMode {
+	// default to a typical mode value
+	mode := os.FileMode(0o755)
+	if a.IsDir() {
+		mode |= os.ModeDir
+	}
+	// the underlying implementation for symlinks and hardlinks share the same semantics in the tree implementation
+	// (meaning resolution is required) where as in a real file system this is taken care of by the driver
+	// by making the file point to the same inode as another --making the indirection transparent to applications.
+	if a.Node.FileType == file.TypeSymlink || a.Node.FileType == file.TypeHardLink {
+		mode |= os.ModeSymlink
+	}
+	return mode
+}
+
+// ModTime returns a dummy value. Traditionally would be the modification time for the given file.
+func (a *fileinfoAdapter) ModTime() time.Time {
+	return time.Now()
```

Comment:
(Nit) Interesting... if the return value really doesn't matter for our code, and we know we won't have a "real" value to offer to the consumer, I might suggest making this implementation deterministic to avoid the potential for bugs in future consumption of this method.

Reply:
in actuality any usage of this function should be considered unsupported. Should probably even replace these with panics?

Reply:
I've adjusted all unused methods that are there to fulfill the interface with panic statements.

Reply:
I think that's a good solution 👍